tests/rptest/redpanda_cloud_tests/high_throughput_test.py

```diff
@@ -489,7 +489,7 @@ def test_max_connections(self):
         # Account for the metadata traffic of 1%
         _target_total = int(self._advertised_max_client_count)
         _target_per_node = int(_target_total // len(self.cluster.nodes))
-        _target_total = int(_target_total * 0.99)
+        _target_total = int(_target_total * 0.9)
```

Comment:
dropping to 90% seems aggressive to me, but i don't have a clear understanding of the metadata overhead. i think it would be more conservative to drop to 95% because i have a hard time believing we have that much metadata overhead.

also, all the comment about metadata needs to be update

Reply:
Yes, this is not good. I've updated the test to actually push through all of the connections and detect how much concurrent ones can be achieved. See next commit